### ISSUE TITLE
macOS: implement time.thread_time() using thread_info()

### ISSUE THREAD

**Issue by vstinner** [python/cpython#76274]
BPO | [32093](https://bugs.python.org/issue32093)
--- | :---
Nosy | @ronaldoussoren, @pitrou, @vstinner, @ned-deily

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2018-09-19.23:51:04.999>
created_at = <Date 2017-11-20.17:20:53.186>
labels = ['OS-mac', '3.7']
title = 'macOS: implement time.thread_time() using thread_info()'
updated_at = <Date 2018-09-19.23:51:04.997>
user = 'https://github.com/vstinner'
```

bugs.python.org fields:
```python
activity = <Date 2018-09-19.23:51:04.997>
actor = 'vstinner'
assignee = 'none'
closed = True
closed_date = <Date 2018-09-19.23:51:04.999>
closer = 'vstinner'
components = ['macOS']
creation = <Date 2017-11-20.17:20:53.186>
creator = 'vstinner'
dependencies = []
files = []
hgrepos = []
issue_num = 32093
keywords = []
message_count = 3.0
messages = ['306569', '306704', '325831']
nosy_count = 4.0
nosy_names = ['ronaldoussoren', 'pitrou', 'vstinner', 'ned.deily']
pr_nums = []
priority = 'normal'
resolution = 'wont fix'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue32093'
versions = ['Python 3.7']
```

</p></details>

**Comment by vstinner** [comment]
bpo-32025 added time.thread_time() function. On macOS, CLOCK_THREAD_CPUTIME_ID is not available. I propose to add a macOS implementation using thread_info(): return the sum of user and system times.

**Comment by ronaldoussoren** [comment]
CLOCK_THREAD_CPUTIME_ID should be available on macOS 10.12 and later (which is also when clock_gettime was introduced).

I'm not sure if it is worthwhile to implement this fairly specialised function for older macOS releases due to the maintenance cost.
